### DISCUSSION TITLE
How to disable langsmith tracing for a specific llm call site without disabling other concurrent calls?

### DISCUSSION THREAD

**Discussion by p-barman** [langchain-ai/langsmith-sdk#1357] [Q&A]
![image](https://github.com/user-attachments/assets/53b40684-038c-47ff-90d0-500035e06103)

Currently using` os.environ['LANGCHAIN_TRACING_V2'] = 'false' ` to disable logging for an llm call.
However, I noticed that doing so also disables logging for the time framer - of the other concurrent llm calls happening in FastAPI server.

Any suggestions?

**Comment by hinthornw** [comment]
Mostly like this:
```
with tracing_context(enabled=False)
   ...
```


I can double check whether this is takes precedence if it's mid-trace but that seems to not be relevant here.

We have a few docs on this, one being:
https://docs.smith.langchain.com/observability/how_to_guides/tracing/trace_without_env_vars

Ya toggling global environment in a process is bad practice and we cache it anyway so it wouldn't work.

**Reply by p-barman** [reply]
Appreciate the suggestion but did not resolve my issue:

wrapping the callsite with `tracing_context(enabled=False):` still logged my run to langsmith!

Open to any more suggestions/debugging tips!

<img width="756" alt="image" src="https://github.com/user-attachments/assets/04b719e9-dca8-4576-a967-e4b75b3e67b9" />

**Reply by hinthornw** [reply]
If it's mid-langchain trace we don't support that

**Reply by vikram-typeset** [reply]
I am also facing the same issue. 

I am not able to disable the tracing of LLM calls inside a ToolNode.

Any suggestion?
